src/lightning_lite/plugins/collectives/torch_.py

```diff
@@ -133,6 +135,12 @@ def setup(
 
     def teardown(self) -> Self:  # type: ignore[valid-type]
         super().teardown()
+        default_group = dist.GroupMember.WORLD
+        if default_group is not None:  # ensures it hasn't been destroyed already
+            group_map = dist.distributed_c10d._pg_map
+            if len(group_map) == 1 and default_group in group_map:  # ensures that only the default group is left
+                self.destroy_group(default_group)
```

Comment:
This will destroy the default process group (created internally by `init_process_group`) only if it hasn't been destroyed already and it's the last group leftover

Reply:
Let's not manipulate the default group. A collective operates only on one group and should not teardown another one. I have been puzzled from the beginning why we are managing the default group inside the collectives.

Reply:
`collective.setup()` will call `torch.distributed.init_process_group()` which internally manipulates the default group.

This is why I added its destruction to `collective.teardown()`

The only alternative here is that this class doesn't call `init_process_group` for the user at all, then we would either create a separate utility/class for that or just call this boilerplate manually somewhere else

Reply:
I prefer to have it the current way.

Just for the record: We could also think about not destroying the default group ever as this would be handled automatically upon the process termination. 
And if we don't send anything over that connection, the only downside of having the group inplace would be an open socket on each process, right?